discovery/gossiper.go

```diff
@@ -343,22 +344,27 @@ func New(cfg Config, selfKey *btcec.PublicKey) *AuthenticatedGossiper {
 // specified, then the update will be applied to all outgoing channels from the
 // source node. Policy updates are done in two stages: first, the
 // AuthenticatedGossiper ensures the update has been committed by dependent
-// sub-systems, then it signs and broadcasts new updates to the network.
+// sub-systems, then it signs and broadcasts new updates to the network. A
+// mapping between outpoints and updated channel policies is returned, which is
+// used to update the forwarding policies of the underlying links.
 func (d *AuthenticatedGossiper) PropagateChanPolicyUpdate(
-	newSchema routing.ChannelPolicy, chanPoints ...wire.OutPoint) error {
+	newSchema routing.ChannelPolicy, chanPoints ...wire.OutPoint) (
+	map[wire.OutPoint]*channeldb.ChannelEdgePolicy, error) {
 
 	errChan := make(chan error, 1)
+	chanPolicyChan := make(chan map[wire.OutPoint]*channeldb.ChannelEdgePolicy, 1)
```

Comment:
I'd make a struct containing the map and the error and use only a single channel.